app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,54 @@
+class Admin::UsersController < ApplicationController
+  before_action :find_user, only: [:edit, :update, :destroy, :show]
+  before_action :admin_required
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to admin_user_path(current_user.id)
+    else
+      render :new
+    end
+  end
+
+  def update
+    @user.admin = true
+    if @user.save(validate: false)
+      redirect_to admin_user_path(current_user.id)
+    else
+      @users = User.all.includes(:tasks)
+      render :show
+    end
+  end
+
+  def destroy
+    @user.admin = false
+    if @user.save(validate: false)
+      redirect_to admin_user_path(current_user.id)
+    else
+      @users = User.all.includes(:tasks)
+      render :show
+    end
+  end
+
+  def show
+    @users = User.all.id_asc.includes(:tasks)
+  end
+
+  private 
+  def find_user
+    @user = User.find(params[:id])
+  end
+
+  def user_params
+    params.require(:admin).permit(:name, :email, :admin, :password, :password_confirmation)
+  end
+
+  def admin_required
+    redirect_to user_path(current_user.id) unless current_user.admin
```

Comment:
booleanの判定メソッドの結果を判定する形式の方が分かりやすいかと思います。
```suggestion
    redirect_to user_path(current_user.id) unless current_user.admin?
```

Reply:
65034d5
にて修正いたしました